packages/@ember/-internals/routing/lib/system/route.ts

```diff
@@ -2558,6 +2588,12 @@ Route.reopen(ActionHandler, Evented, {
         }
       }
 
+      // Some QPs have been updated, and those changes need to be propogated
+      // immediately. Eventually, we should work on making this async somehow.
+      if (EMBER_METAL_TRACKED_PROPERTIES && qpUpdated === true) {
+        flushAsyncObservers(false);
```

Comment:
It is not clear to me at all why we need to sync flush here. Can you make an issue for us to track removing this (likely evaluating the specific failing test that caused it to be made sync)?

Reply:
For reference, the failing tests were:

```
<LinkTo /> component with query params (routing): the <LinkTo /> component applies activeClass when query params are not changed
{{link-to}} component with query params (routing): the {{link-to}} applies activeClass when query params are not changed
```